src/snowcli/cli/snowpark/common.py

```diff
@@ -132,6 +132,45 @@ def execute(self, execution_identifier: str) -> SnowflakeCursor:
     def artifact_stage_path(identifier: str):
         return generate_deploy_stage_name(identifier).lower()
 
+    def create_query(
+        self,
+        identifier: str,
+        return_type: str,
+        handler: str,
+        artifact_file: str,
+        packages: List[str],
+        external_access_integrations: Optional[List[str]] = None,
+        secrets: Optional[Dict[str, str]] = None,
+        execute_as_caller: bool = False,
+    ) -> str:
+        packages_list = ",".join(f"'{p}'" for p in packages)
+        query = [
+            f"create or replace {self._object_type} {identifier}",
+            f"returns {return_type}",
+            "language python",
+            "runtime_version=3.8",
+            f"imports=('{artifact_file}')",
+            f"handler='{handler}'",
+            f"packages=({packages_list})",
+        ]
+
+        if external_access_integrations:
+            external_access_integration_name = ",".join(
+                f"{e}" for e in external_access_integrations
+            )
+            query.append(
+                f"external_access_integrations=({external_access_integration_name})"
+            )
+
+        if secrets:
```

Comment:
don't you need to have both? Can you provide only secret without integration? Or the other way around? Should we check it?

Reply:
You can create integration with no secret, you don't need any credentials to public service